src/tiled/objectselectiontool.cpp

```diff
@@ -551,8 +586,25 @@ void ObjectSelectionTool::mouseReleased(QGraphicsSceneMouseEvent *event)
             break;
         }
         const Qt::KeyboardModifiers modifiers = event->modifiers();
+        QSet<MapObjectItem*> selection = mapScene()->selectedObjectItems();
+        if (modifiers & Qt::AltModifier) {
+            auto underlyingObjects = objectItemsAt(event->scenePos());
+
+            if (underlyingObjects.isEmpty())
+                break;
+            auto currRotation = std::find(underlyingObjects.begin(),
+                                          underlyingObjects.end(), mSelectedInUnderlyingRotation);
```

Comment:
I tried this approach and realized that it is not compatible with a shift selection. If you start Shift+Alt+Click'ing in the middle of the stack, once bottom-most element is included to the selection, `nextItem` will always be the first element in the stack instead of continue iterating.
Also it is not compatible with a shift **un**selection - in this case we need to access the next selected object.